packaging/snapd.mk

```diff
@@ -68,7 +68,9 @@ all: $(go_binaries)
 $(builddir)/snap: GO_TAGS += nomanagers
 $(builddir)/snap $(builddir)/snap-seccomp $(builddir)/snapd-apparmor:
 	go build -o $@ $(if $(GO_TAGS),-tags "$(GO_TAGS)") \
-		-buildmode=pie -ldflags=-w -mod=vendor \
+		-buildmode=pie \
+		-ldflags="-w -B 0x$$(head -c20 /dev/urandom|od -An -tx1|tr -d ' \n')" \
```

Comment:
I see your comment about openSUSE 15.5. I think we should test the version of go, and disable completely build-id id on older versions.

Reply:
This only affects the GNU build ID note in the binary, not the Go one. And yes, not all Go toolchains we build with support `-B gobuildid`, specifically the one available on openSUSE 15.5 did not, but there could be others too (we only use snapd.mk in RPM spec for openSUSE). I'll add a note to fix this in the future.

Reply:
Do we need to have build id for distros that do not support it?

Reply:
Is not it go 1.22 on openSUSE 15.5? It sounds to me like a bug if it does not support it. Is there a bug on openSUSE that tracks this issue?

Reply:
No, 15.5 is a fixed release. The default go is 1.19, the latest available directly from default repositories is 1.20, `gobuildid` was only added in 1.22.

Reply:
Come to think of it, can we do something else instead? Remember that bit about RPM and reproducible builds? Could we use a build ID that is derived from the reproducible build epoch? I cannot find a link but I think withuot this, it would be a regression in terms of how we build reproducibly.